docs/src/main/asciidoc/stork-registration.adoc

```diff
@@ -0,0 +1,171 @@
+
+== Overview
+
+SmallRye Stork supports service registration backends like Consul and Eureka.
+Traditionally, developers link:{quickstarts-tree-url}/stork-programmatic-custom-registration-quickstart[had to explicitly configure and manually register] their service instances using code.
+With this feature, Quarkus applications can automatically register with a supported registry (e.g., Consul) during startup and deregister during shutdown — without writing boilerplate code.
+
+**IMPORTANT**: Note that automatic registration is possible **only** for built-in registrars such as Consul, Eureka, and Static.
+If you're using a custom registrar, registration must still be handled programmatically.
+Check the `stork-programmatic-custom-registration-quickstart` link:{quickstarts-tree-url}/stork-programmatic-custom-registration-quickstart[directory] example for this use case.
+
+This integration minimizes configuration effort and adheres to Quarkus' philosophy of doing as much as possible at build time.
+
+== How It Works
+
+When the application starts and the `stork-service-registration-consul` dependency, besides the `quarkus-smallrye-stork`, is present:
+
+- Quarkus performs a build-time check for registration configuration.
+- If valid configuration is detected (or can be inferred), registration metadata is generated.
+- The application registers itself automatically using the detected or configured IP address and port.
+- During shutdown, the service is automatically deregistered.
+
+The solution is located in the `stork-automatic-consul-registration-quickstart` link:{quickstarts-tree-url}/stork-automatic-consul-registration-quickstart[directory].
+
+
+== Configuration Options
+
+The registration system uses sensible defaults but also allows for fine-grained control per service.
+
+|===
+|Property |Description |Default
+
+|`quarkus.stork.<service-name>.service-registrar.type`
+|Backend type (`consul`, etc.)
+|Auto-detected if only one registrar is used
+
+|`quarkus.stork.<service-name>.service-registrar.ip-address`
+|IP address to register
+|Auto-detected using the host's first non-loopback address
+
+|`quarkus.stork.<service-name>.service-registrar.port`
+|Port to register
+|`quarkus.http.port`
+
+|`quarkus.stork.<service-name>.service-registrar.service-id`
+|Consul-specific service ID
+|Generated UUID
+
+|`quarkus.stork.<service-name>.service-registrar.enabled`
+|Enable or disable registration
+|`true`
+
+|`quarkus.stork.<service-name>.service-registrar.parameters.health-check-url`
+|Health check endpoint
+|If available, uses `liveness` endpoint
+
+|`quarkus.stork.<service-name>.service-registrar.parameters.health-check-interval`
+|How often Consul pings for health
+|Not set
+
+|`quarkus.stork.<service-name>.service-registrar.parameters.health-check-deregister-after`
+|Time after which the service is deregistered if unhealthy
+|Not set
+|===
```

Comment:
Shouldn't you use our automatic config doc generation instead? It's going to be annoying to maintain and error prone.